cmds/exp/getty/getty.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// getty Open a TTY and invoke a shell
+// There are no special options and no login support
+// Also getty exits after starting the shell so if one exits the shell, there
+// is no more shell!
+//
+// Synopsis:
+//     getty <port> <baud> [term]
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/u-root/u-root/pkg/termios"
+	"github.com/u-root/u-root/pkg/upath"
+)
+
+var (
+	verbose = flag.Bool("v", false, "verbose log")
+	debug   = func(string, ...interface{}) {}
+	cmdList []string
+	envs    []string
+)
+
+func init() {
+	r := upath.UrootPath
+	cmdList = []string{
+		r("/bin/defaultsh"),
+		r("/bin/sh"),
+	}
+}
+
+func main() {
+	flag.Parse()
+
+	if *verbose {
+		debug = log.Printf
+	}
+
+	port := flag.Arg(0)
+	baud, err := strconv.Atoi(flag.Arg(1))
+	if err != nil {
+		baud = 0
+	}
+	term := flag.Arg(2)
+
+	ttyS, err := termios.NewTTYS(port)
+	if err != nil {
+		debug("Unable to open port %s: %v", port, err)
+		return
+	}
+
+	if _, err := ttyS.Serial(baud); err != nil {
```

Comment:
I would suggest you print a message here and continue. After all, if you are printing a message, then the user should be able to see it, which means the baud rate is probably fine. 

I would even argue that setting the baud rate is probably not a great idea, there is so much to go wrong and it is almost certainly correct?

Reply:
Well that one is harder ... the log will go to the console, not the serial port (it will certainly be different otherwise this command is useless).
Not setting the baudrate is possible if the parameter is 0. But I also changed `Serial` to make sure the other parameters are set even is the baudrate is wrong !

Reply:
done